weaviate/collections/classes/generative.py

```diff
@@ -0,0 +1,925 @@
+
+        See the [documentation](https://weaviate.io/developers/weaviate/modules/reader-generator-modules/generative-openai)
+        for detailed usage.
+
+        Arguments:
+            `api_version`
+                The API version to use. Defaults to `None`, which uses the server-defined default
+            `base_url`
+                The base URL where the API request should go. Defaults to `None`, which uses the server-defined default
+            `deployment_id`
+                The deployment ID to use. Defaults to `None`, which uses the server-defined default
+            `frequency_penalty`
+                The frequency penalty to use. Defaults to `None`, which uses the server-defined default
+            `max_tokens`
+                The maximum number of tokens to generate. Defaults to `None`, which uses the server-defined default
+            `model`
+                The model to use. Defaults to `None`, which uses the server-defined default
+            `presence_penalty`
+                The presence penalty to use. Defaults to `None`, which uses the server-defined default
+            `resource_name`
+                The name of the OpenAI resource to use. Defaults to `None`, which uses the server-defined default
+            `stop`
+                The stop sequences to use. Defaults to `None`, which uses the server-defined default
+            `temperature`
+                The temperature to use. Defaults to `None`, which uses the server-defined default
+            `top_p`
+                The top P to use. Defaults to `None`, which uses the server-defined default
+            `images`
+                Any query-specific external images to use in the generation. Passing a string will assume a path to the image file and, if not found, will be treated as a base64-encoded string.
+                The number of images passed to the prompt will match the length of this field.
+            `image_properties`
+                Any query-specific internal image properties to use in the generation. These will be sourced from the object's properties.
+                The number of images passed to the prompt will match the value of `limit` in the search query.
+
+        """
+        return _GenerativeOpenAI(
+            api_version=api_version,
+            base_url=base_url,
+            deployment_id=deployment_id,
+            frequency_penalty=frequency_penalty,
+            max_tokens=max_tokens,
+            model=model,
+            presence_penalty=presence_penalty,
+            resource_name=resource_name,
+            stop=stop,
+            temperature=temperature,
+            top_p=top_p,
+            is_azure=False,
+            images=(parse_blob(image) for image in images) if images is not None else None,
+            image_properties=image_properties,
+        )
+
+    @staticmethod
+    def openai_azure(
```

Comment:
for the future, how could we add image-generation here without breaking anything? (also for the output)

Reply:
I think we would either expose another field that defines which type of generation to do, e.g. `generation_type: Literal["text", "image", "all"]`, and then have distinct fields in the output for text and image